cmd/dendrite-monolith-server/main.go

```diff
@@ -55,7 +55,7 @@ func main() {
 	deviceDB := base.CreateDeviceDB()
 	keyDB := base.CreateKeyDB()
 	federation := base.CreateFederationClient()
-	keyRing := keydb.CreateKeyRing(federation.Client, keyDB)
+	keyRing := keydb.CreateKeyRing(federation.Client, keyDB, cfg)
```

Comment:
Please don't do this. I don't like this because it's not obvious what `CreateKeyRing` actually requires from `cfg`, which damages readability and damages the ability to independently test that component. I can clearly see in this diff why you are adding it, but once we lose that context on merge it'll be too late. Please yank out `KeyPerspectives` into a separate struct and require that instead.

Reply:
Split it out.